File: Python/Problems/Problems/problem1375.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1375(Problem):
    def test(self) -> bool:
        flips = [3, 2, 4, 1, 5]

        expected = 2

        result = self.numTimesAllBlue(flips)

        print(result)

        return result == expected

    def numTimesAllBlue(self, flips: List[int]) -> int:
        n = len(flips)
        result = 0
        last = -1
        for i in range(n):
            last = max(last, flips[i])
            if last == i + 1:
                result += 1

        return result

File: C++/Problems/Problems/problem_1375.cpp
#include "problem_1375.h"

bool problem_1375::test() {
    std::vector flips { 3, 2, 4, 1, 5 };

    int expected = 2;

    auto result = numTimesAllBlue(flips);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1375::numTimesAllBlue(std::vector<int> &flips) {
    int n = flips.size();
    int result = 0;
    int last = -1;
    for (int i = 0; i < n; ++i)
    {
        last = std::max(last, flips[i]);
        if (last == i + 1)
            result++;
    }

    return result;
}
